### ISSUE TITLE
0.4.0 .btn-primary hover effect is inconsistent with other buttons

### ISSUE THREAD

**Issue by acoard** [bcgov/mygovbc-bootstrap-theme#4]
**Example**

https://bcgov.github.io/mygovbc-bootstrap-theme/docs/index.html#buttons

Hover over the "Primary" button (or any of the Large/Default/Small/Mini buttons).

**Description**

I recently upgraded to 0.4.0. [ In the 0.4.0 changelog, it says the default and primary button colours were changed.](https://github.com/bcgov/mygovbc-bootstrap-theme/releases/tag/v0.4)  However, it appears the background colour on hover for these buttons have not been changed.  The current colour is `#fdc94b`, a bright yellow.

`.btn-default` has the same background colour, but on :hover the colour changes to `#294266`, a darker shade of the new colour.

** Expected Behaviour **

`.btn-primary` has a dark blue hover colour, similar to `.btn-default`

----

Lastly, just a quick question. Is there a reason why `.btn-default` and `.btn-primary` are effectively becoming identical while remaining in two classes? I know of at least two projects which make use of both `.btn-default`  and `.btn-primary` and will likely continue to. Of course we can always re-implement `.btn-default` in the project styling, so it's not a big deal either way.

Personally I'd vote for keeping the two separate and distinct, but if they're going to be made to look identical then why not combine the classes and remove one? I'd imagine removing a CSS class is a breaking change, so it's completely understandable if you hold off for a bit.

Anyways, thanks for all the hard work! I greatly appreciate what you guys are doing here.  Excluding the above the upgrade to 0.4.0 has gone without a snag.

Cheers!

Adam

**Comment by GregTurner** [comment]
@acoard thanks for finding and reporting this, I think the :hover should be #294266 for .btn-default instead of that yellow.  

Re: the btn.default = btn.primary not sure why that was decided, but it's worth re-considering when doing the bootstrap v4 of this theme.

**Comment by acoard** [comment]
Happy to help take part in this wonderful thing called open source. ;)

Glad to hear you guys are willing to reconsider .btn-default not being
identical to .btn-primary.  I still strongly support having the previous
.btn-default re-instated. It's lack of colour made it perfect for cases
where the button didn't warrant much visual weight (e.g. "Print" or "Back"
buttons). In most cases we use .btn-primary to indicate the primary action
the user will take on that page, and .btn-default for optional or
infrequent actions.

I'll make sure to keep an eye out for when you folks are doing the
bootstrap v4 upgrade and can make the case there, if appropriate.

**Comment by acoard** [comment]
For anyone else coming to this thread and looking to get the old .btn-default stylings while waiting for the fix, here's the SCSS:

```
.btn-default {
  color: #333;
  background-color: #f4f6f9;
  border-color: #ccc;
  &:hover {
    color: #333;
    background-color: #d3dbe7;
    border-color: #adadad;
  }
}

//Bonus, .btn-primary hover colour:
.btn-primary:hover {
  background-color: #294266;
}
```

**Comment by GregTurner** [comment]
Fixed in v0.4.1

**Comment by GregTurner** [comment]
FYI, there's no  `btn-default` in Bootstrap V4, therefore it won't be an issue going forward.
